docs/pages/enroll-resources/database-access/enroll-google-cloud-databases/alloydb.mdx

```diff
@@ -0,0 +1,307 @@
+
+```ini
+alloydb.instances.connect
+alloydb.users.login
+serviceusage.services.use
+```
+
+For least privilege, create and assign a custom IAM role containing exactly the permissions above. Alternatively, you may grant these [predefined roles](https://cloud.google.com/alloydb/docs/reference/iam-roles-permissions):
+
+* Cloud AlloyDB Database User (`roles/alloydb.databaseUser`)
+* Cloud AlloyDB Client (`roles/alloydb.client`)
+* [Service Usage Consumer (`roles/serviceusage.serviceUsageConsumer`)](https://cloud.google.com/service-usage/docs/access-control#serviceusage.serviceUsageConsumer)
+
+![Grant permissions to user Service Account](../../../../img/database-access/guides/alloydb/user-service-account-permissions.png)
+
+Click "Done".
+
+### Grant access to the service account
+
+The Teleport Database Service must be able to impersonate this service account.
+Navigate to the "alloydb-user" service account overview page and select the
+"Principals with Access" tab:
+
+![Select Principals with Access Tab](../../../../img/database-access/guides/alloydb/user-service-account-principals-with-access.png)
+
+Click "Grant Access" and add the "teleport-db-service" principal ID.
+Select the "Service Account Token Creator" role and save the change:
+
+![Grant Service Account Token Creator to Database Service](../../../../img/database-access/guides/alloydb/user-service-account-grant-access.png)
+
+<Admonition type="note" title="Service account permissions">
+  The "Service Account Token Creator" IAM role includes more permissions than
+  the Teleport Database Service needs. To further restrict the service 
+  account, you can create a role that includes only the following permission:
+  ```ini
+  # Used to generate IAM auth tokens when connecting to a database instance.
+  iam.serviceAccounts.getAccessToken
+  ```
+</Admonition>
+
+## Step 3/9. Configure your AlloyDB database
+
+Teleport uses [IAM database authentication](https://cloud.google.com/alloydb/docs/database-users/manage-iam-auth)
+with AlloyDB instances.
+
+Navigate to your instance settings and enable the `alloydb.iam_authentication` flag under Advanced Configuration Options section.
+
+![Enable IAM Authentication](../../../../img/database-access/guides/alloydb/flag-iam-authentication-on.png)
+
+### Create a database user
+
+Now go back to the Users page of your AlloyDB instance and add a new user
+account. In the sidebar, choose "Cloud IAM" authentication type and add the
+"alloydb-user" service account that you created in
+[the second step](#step-29-create-a-service-account-for-a-database-user):
+
+![Add AlloyDB User Account](../../../../img/database-access/guides/alloydb/add-user-account.png)
+
+Press "Add" and your Users table should look similar to this:
+
+![AlloyDB User Accounts Table](../../../../img/database-access/guides/alloydb/user-account-added.png)
+
+See [Manage IAM authentication](https://cloud.google.com/alloydb/docs/database-users/manage-iam-auth)
```

Comment:
What additional information would the user expect to find here? If the goal is to indicate to the user that these docs exist, and isn't required to get the user to step 4, I think it makes sense to move it to the "Next steps" section.